src/test/record/test_frequency.py

```diff
@@ -0,0 +1,389 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from caret_analyze.record import ColumnValue
+from caret_analyze.record.frequency import Frequency
+from caret_analyze.record.record_factory import RecordFactory, RecordsFactory
+
+
+def create_records(records_raw, columns):
+    records = RecordsFactory.create_instance()
+    for column in columns:
+        records.append_column(column, [])
+
+    for record_raw in records_raw:
+        record = RecordFactory.create_instance(record_raw)
+        records.append(record)
+    return records
+
+
+def to_dict(records):
+    return [record.data for record in records]
+
+
+class TestFrequencyRecords:
+
+    def test_callback_empty_case(self):
+        records_raw = [
+        ]
+        columns = [ColumnValue('/node/callback/callback_start_timestamp'),
+                   ColumnValue('/node/callback/callback_end_timestamp')]
+        records = create_records(records_raw, columns)
+
+        callback_frequency = Frequency(records, target='callback')
+
+        expect_raw = [
+        ]
+        result = to_dict(callback_frequency.to_records(
+            initial_timestamp=1000000000))
+        assert result == expect_raw
+
+    def test_callback_normal_case(self):
+        records_raw = [
+            {'/node/callback/callback_start_timestamp': 1000000000,
+             '/node/callback/callback_end_timestamp': 1000001000},
+            {'/node/callback/callback_start_timestamp': 1000500000,
+             '/node/callback/callback_end_timestamp': 1000501000},
+            {'/node/callback/callback_start_timestamp': 2000000000,
+             '/node/callback/callback_end_timestamp': 2000001000},
+            {'/node/callback/callback_start_timestamp': 2000300000,
+             '/node/callback/callback_end_timestamp': 2000301000},
+            {'/node/callback/callback_start_timestamp': 2000500000,
+             '/node/callback/callback_end_timestamp': 2000501000},
+        ]
+        columns = [ColumnValue('/node/callback/callback_start_timestamp'),
+                   ColumnValue('/node/callback/callback_end_timestamp')]
+        records = create_records(records_raw, columns)
+
+        callback_frequency = Frequency(records, target='callback')
+
+        expect_raw = [
+            {'callback_start_timestamp': 1000000000, 'frequency': 2},
+            {'callback_start_timestamp': 2000000000, 'frequency': 3}
+        ]
+        result = to_dict(callback_frequency.to_records(
+            initial_timestamp=1000000000))
```

Comment:
引数を以下にお願いします。
```suggestion
        result = to_dict(callback_frequency.to_records())
```
こちらも、周波数であれば1s毎が普通ですので、デフォルト値を設定して置いたほうが良いです。
```
def to_records(self, interval_ns = 1000000000):
pass
```

Reply:
interval_nsに対して、以下のようなケースをテストしておいた方が安心です。
- デフォルト値の1s
- 引数値をか得た場合の0.1s（1s以外ならOK）


↓の三角測量にあたります。
https://kakakakakku.hatenablog.com/entry/2019/09/24/085131

Reply:
ご指摘ありがとうございます。修正させて頂きました。

指摘して頂いた事項の他に、
例えば他のコールバックと、基準となるタイムスタンプを統一したい場合に initial_timestamp の引数は必要だと思うので、以下のような実装を考えています。
デフォルトでは、target_column の最も早い timestamp を使用します
```
def to_records(self, initial_timestamp: Optional[int], interval_ns: Optional[int] = 1000000000):
    pass
```

Reply:
はい、良いと思います。
”基準”とすると、”最初”のinitialより、base_timestampではいかがでしょうか？